src/manifolds/ProbabilitySimplex.jl

```diff
@@ -0,0 +1,352 @@
+    error("The injectivity radius on $(M) depends on the point and has no closed form globally")
+end
+function injectivity_radius(M::ProbabilitySimplex, ::AbstractRetractionMethod)
+    error("The injectivity radius on $(M) depends on the point and has no closed form globally")
+end
+injectivity_radius(M::ProbabilitySimplex, p, ::ExponentialRetraction) = injectivity_radius(M,p)
+injectivity_radius(M::ProbabilitySimplex, p, ::SoftmaxRetraction) = injectivity_radius(M,p)
+
+@doc raw"""
+    inner(M::ProbabilitySimplex,p,X,Y)
+
+Compute the inner product of two tangent vectors `X`, `Y` from the tangent space $T_pΔ^n$ at
+`p`. The formula reads
+````math
+g_p(X,Y) = \sum_{i=1}^{n+1}\frac{X_iY_i}{p}
+````
+"""
+function inner(::ProbabilitySimplex, p, X, Y)
+    d = zero(Base.promote_eltype(p,X,Y))
+    @inbounds for i in eachindex(p,X,Y)
+        d += X[i]*Y[i]/p[i]
+    end
+    return d
+end
+
+@doc raw"""
+    inverse_retract(M::ProbabilitySimplex, p, q, ::SoftmaxInverseRetraction)
+
+Compute a first order approximation by projection. The formula reads
+````math
+\operatorname{retr}^{-1}_p q = \bigl( I - \frac{1}{n}\mathbb{1}\mathbb{1}^\mathrm{T}\bigr)(\log(q)-\log(p))
+````
+where $\mathbb{1}$ is the column vector containing ones and $\log$ is applied elementwise.
+"""
+inverse_retract(::ProbabilitySimplex, ::Any, ::Any, ::SoftmaxInverseRetraction)
+
+function inverse_retract!(::ProbabilitySimplex{n}, X, p, q, ::SoftmaxInverseRetraction) where {n}
+    X .= log.(q) .- log.(p)
+    meanlogdiff = mean(X)
+    X .-= meanlogdiff
+    return X
+end
+
+
+@doc raw"""
+    log(M::ProbabilitySimplex, p, q)
+
+Compute the logarithmic map of `p` and `q` on the [`ProbabilitySimplex`](@ref) `M`.
+
+````math
+\log_pq = \frac{d_{Δ^n}(p,q)}{\sqrt{1-⟨\sqrt{p},\sqrt{q}⟩}}(\sqrt{pq} - ⟨\sqrt{p},\sqrt{q}⟩p),
+````
+
+where $pq$ and $\sqrt{p}$ is meant elementwise.
+"""
+log(::ProbabilitySimplex, ::Any...)
+
+function log!(M::ProbabilitySimplex, X, p, q)
+    if p ≈ q
+        fill!(X,0)
+    else
+        z = sqrt.(p .* q)
+        s = sum(z)
+        X .= 2acos(s) / sqrt(1 - s^2) .* (z .- s .* p)
+    end
+    return X
+end
+
+@doc raw"""
+    manifold_dimension(M::ProbabilitySimplex{n})
+
+Returns the manifold dimension of the probability simplex in $ℝ^{n+1}$, i.e.
+````math
+    \dim_{Δ^n} = n.
+````
+"""
+manifold_dimension(::ProbabilitySimplex{n}) where {n} = n
+
+"""
+    mean(
+        S::ProbabilitySimplex,
+        x::AbstractVector,
+        [w::AbstractWeights,]
+        method = GeodesicInterpolationWithinRadius(a);
+        kwargs...,
+    )
+
+Compute the Riemannian [`mean`](@ref mean(M::Manifold, args...)) of `x` using
+[`GeodesicInterpolationWithinRadius`](@ref), where the minimum of the injectivity radii
+of the points in `x` is used as the value `a`.
+"""
+mean(::ProbabilitySimplex, ::Any...)
+
+function mean!(M::ProbabilitySimplex, p, x::AbstractVector, w::AbstractVector; kwargs...)
+    a = a = min(injectivity_radius.(Ref(M),x)...)
```

Comment:
As I mentioned in my comment, we don't need a radius here: https://github.com/JuliaManifolds/Manifolds.jl/pull/151#issuecomment-610866602

Reply:
Ok.